user/api_test.go

```diff
@@ -66,33 +77,61 @@ var (
 	/*
 	 * expected path
 	 */
-	logger      = log.New(os.Stdout, USER_API_PREFIX, log.LstdFlags|log.Lshortfile)
-	mockStore   = NewMockStoreClient(FAKE_CONFIG.Salt, false, false)
-	mockMetrics = highwater.NewMock()
-	shoreline   = InitApiTest(FAKE_CONFIG, logger, mockStore, mockMetrics)
+	logger             = log.New(os.Stdout, USER_API_PREFIX, log.LstdFlags|log.Lshortfile)
+	mockStore          = NewMockStoreClient(FAKE_CONFIG.Salt, false, false)
+	mockMetrics        = highwater.NewMock()
+	mockMarketoManager = NewTestManager()
+	shoreline          = InitAPITest(FAKE_CONFIG, logger, mockStore, mockMetrics, mockMarketoManager)
 	/*
 	 *
 	 */
 	mockNoDupsStore = NewMockStoreClient(FAKE_CONFIG.Salt, true, false)
-	shorelineNoDups = InitApiTest(FAKE_CONFIG, logger, mockNoDupsStore, mockMetrics)
+	shorelineNoDups = InitAPITest(FAKE_CONFIG, logger, mockNoDupsStore, mockMetrics, mockMarketoManager)
 	/*
 	 * failure path
 	 */
 	mockStoreFails = NewMockStoreClient(FAKE_CONFIG.Salt, false, MAKE_IT_FAIL)
-	shorelineFails = InitApiTest(FAKE_CONFIG, logger, mockStoreFails, mockMetrics)
+	shorelineFails = InitAPITest(FAKE_CONFIG, logger, mockStoreFails, mockMetrics, mockMarketoManager)
 
 	responsableStore      = NewResponsableMockStoreClient()
 	responsableGatekeeper = NewResponsableMockGatekeeper()
 	responsableShoreline  = InitShoreline(FAKE_CONFIG, responsableStore, mockMetrics, responsableGatekeeper)
 )
 
 func InitShoreline(config ApiConfig, store Storage, metrics highwater.Client, perms clients.Gatekeeper) *Api {
-	api := InitApiTest(config, logger, store, metrics)
+	api := InitAPITest(config, logger, store, metrics, mockMarketoManager)
 	api.AttachPerms(perms)
 	return api
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// LIKE RECTANGLE what is needed in Area and Perimeter
+type MockUser struct {
+}
+type MockManager struct {
+}
+
+func (U *MockUser) Email() string {
```

Comment:
Do you use this?

Reply:
I dont believe so since it is not needed to create the manager.